hotshot-state-prover/src/service.rs

```diff
@@ -101,7 +100,8 @@ impl ProverServiceState {
         let stake_table = fetch_stake_table_from_sequencer(&config.sequencer_url, None)
             .await
             .with_context(|| "Failed to initialize stake table")?;
-        let st_state = compute_stake_table_commitment(&stake_table, config.stake_table_capacity);
+        let st_state = compute_stake_table_commitment(&stake_table, config.stake_table_capacity)
```

Comment:
will it be clearer to introduce a `StakeTable` wrapper type around `Vec<PeerConfig>`? and implement `.commitment()` as a method. 

we can auto into or deref on the wrapper type. 

if that's too troublesome in your liking, then adding an alias would also help. because it's not straightforward for first comer to realize this type is what we call stake_table.

Reply:
Do you think that we should change (maybe) all occurrences of `Vec<PeerConfig>`?

Reply:
hmm.... good question. @imabdulbasit @Ayiga @bfish713 wdyt? 
I think only when the intention of the caller is clearly expecting a full stake table. there are cases where caller is just getting some slices or a few members of it, in those cases, maybe leave it as `Vec<PeerConfig>`?

Reply:
I also agree that using a wrapper/type alias would be much clearer. But perhaps we need some more discussions on it and do it in another PR.

Reply:
Sure I don't mind another linked issue

Reply:
https://github.com/EspressoSystems/espresso-network/issues/3188

Reply:
I would be in favor of the wrapper type, but it does seem like it can be done seperately